xbmc/platform/darwin/tvos/TopShelf/ServiceProvider.mm

```diff
@@ -33,35 +35,30 @@ - (TVTopShelfContentStyle)topShelfStyle
 
 - (NSArray<TVContentItem*>*)topShelfItems
 {
+  // Retrieve store URL
   __auto_type storeUrl = [tvosShared getSharedURL];
   if (!storeUrl)
     return @[];
+  storeUrl = [storeUrl URLByAppendingPathComponent:@"RA" isDirectory:YES];
 
-  __auto_type const sharedID = [tvosShared getSharedID];
-  __auto_type const shared = [[NSUserDefaults alloc] initWithSuiteName:sharedID];
-  __auto_type topShelfItems = [[NSMutableArray alloc] init];
-  __auto_type wrapperIdentifier =
-      [[TVContentIdentifier alloc] initWithIdentifier:@"shelf-wrapper" container:nil];
-
-  NSArray* movieArray = nil;
-  NSArray* tvArray = nil;
-  NSDictionary* sharedDict = nil;
 
-  if ([tvosShared isJailbroken])
+  // Retrieve shared dict
+  NSDictionary* sharedDict;
+  if (CDarwinEmbedUtils::IsIosSandboxed())
   {
-    __auto_type sharedDictUrl =
-        [storeUrl URLByAppendingPathComponent:@"shared.dict" isDirectory:NO];
-    sharedDict = [NSDictionary dictionaryWithContentsOfFile:[sharedDictUrl path]];
-
-    movieArray = [sharedDict valueForKey:@"movies"];
-    tvArray = [sharedDict valueForKey:@"tv"];
+    __auto_type const sharedID = [tvosShared getSharedID];
```

Comment:
since it's a `.mm` file now, you should use `auto` instead of `__auto_type` (also below)